File: app/src/main/java/com/canli/oya/traininventoryroom/data/dao/CategoryDao.java
package com.canli.oya.traininventoryroom.data.dao;

import android.arch.lifecycle.LiveData;
import android.arch.persistence.room.Dao;
import android.arch.persistence.room.Delete;
import android.arch.persistence.room.Insert;
import android.arch.persistence.room.OnConflictStrategy;
import android.arch.persistence.room.Query;

import com.canli.oya.traininventoryroom.data.entities.CategoryEntry;

import java.util.List;

@Dao
public interface CategoryDao {

   @Query("SELECT * FROM categories")
    LiveData<List<String>> getAllCategories();

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    void insertCategory(CategoryEntry category);

    @Delete
    void deleteCategory(CategoryEntry category);

}

File: app/src/main/java/com/canli/oya/traininventoryroom/data/dao/CategoryDao.kt
package com.canli.oya.traininventoryroom.data.dao

import androidx.room.Dao
import androidx.room.Query
import com.canli.oya.traininventoryroom.data.entities.CategoryEntity
import kotlinx.coroutines.flow.Flow

@Dao
interface CategoryDao : BaseDao<CategoryEntity> {

    @Query("SELECT * FROM categories ORDER BY categoryName")
    fun observeAllCategories(): Flow<List<CategoryEntity>>

    @Query("SELECT * FROM categories ORDER BY categoryName")
    suspend fun getCategoryList() : List<CategoryEntity>

    @Query("SELECT categoryName FROM categories ORDER BY categoryName")
    suspend fun getCategoryNames() : List<String>

    @Query("SELECT * FROM categories WHERE categoryId = :id")
    suspend fun getChosenCategory(id: Int): CategoryEntity
}
